diff --git a/PKGBUILD b/PKGBUILD
index 12f138f..e4c6dba 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -15,10 +15,17 @@ source=("${url}/archive/v${pkgver}.tar.gz"
 sha256sums=('c06a3e5fbe387ed883c0652aece79ea00603e6e6d2a4f54400701807185b7eef'
             '26eb4214d744c16cd4e8976e495f6cad8c7c98d4ffad3ec79b71b6241e0a1bbf')
 
+source+=("fix.patch")
+sha256sums+=("9768e74be7dd6d43c2b2e9f0a02f9bc0b7d3e8998958e5c4c177d3cd540e0892")
+
 build() {
-    jdk="$(find /usr/lib/jvm -maxdepth 1 -regex ".*1[4-9].*" | head -1)" &&
-      test -n "$jdk" && export JAVA_HOME="$jdk"
-    cd "${srcdir}/${pkgname}-${pkgver}"
+    JAVA_HOME="$(find /usr/lib/jvm -maxdepth 1 -regex ".*[0-9]+.*" | sort -Vr | head -1)"
+    if [[ -z $JAVA_HOME ]]; then exit 1; fi
+
+    cd "$pkgname-$pkgver"
+    patch -p 1 < "$srcdir/fix.patch"
+
+    export PATH="$JAVA_HOME/bin:$PATH"
     ./scripts/link_linux.sh
     mvn package -DskipTests
 }
diff --git a/fix.patch b/fix.patch
new file mode 100644
index 0000000..ea34847
--- /dev/null
+++ b/fix.patch
@@ -0,0 +1,13 @@
+diff --git a/src/main/java/org/javacs/ReusableCompiler.java b/src/main/java/org/javacs/ReusableCompiler.java
+index 2d50f74c..fd90abb7 100644
+--- a/src/main/java/org/javacs/ReusableCompiler.java
++++ b/src/main/java/org/javacs/ReusableCompiler.java
+@@ -251,7 +251,7 @@ class ReusableCompiler {
+                 this.context = context;
+             }
+ 
+-            void clear() {
++            public void clear() {
+                 recorded.clear();
+                 sourceMap.clear();
+                 nerrors = 0;
