diff --git a/PKGBUILD b/PKGBUILD
index fcdd708..83b7c5c 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,6 +16,10 @@ provides=(phpactor)
 source=("https://github.com/phpactor/phpactor/archive/refs/tags/${pkgver}.tar.gz")
 sha256sums=("366c3e003272d27b74f07263762dafec33f1f04eeb61bfb800ca8759895cd829")
 
+prepare() {
+  sudo sh -c 'echo "extension=iconv" > /etc/php/conf.d/phpactor.ini'
+}
+
 check() {
   if ! php -m | grep iconv >/dev/null; then
     echo "The iconv extension is not installed for the php installation in use.
